src/core/api/track_choice_manager.ts

```diff
@@ -968,6 +1013,37 @@ export default class TrackChoiceManager {
 
     recursiveUpdateVideoTrack(0);
   }
+
+  /**
+   * Set the adaptation for stream.
+   * If the trick mode is enabled, select the trick mode track if it
+   * exists. If it does not exists, then disable the trick mode before
+   * switching adaptation.
+   * Select the main given adaptation if trick mode is disabled.
+   * @param {Object} adaptation$
+   * @param {Object | null} adaptation
+   */
+  private _setVideoTrack(adaptation$: Subject<Adaptation | null>,
+                         adaptation: Adaptation | null): void {
+    if (adaptation === null) {
+      adaptation$.next(null);
+      return;
+    }
+    const enabledTrickMode = this._trickModeEnabled$.getValue();
+    if (enabledTrickMode) {
+      if (adaptation.trickModeTracks !== undefined &&
+          adaptation.trickModeTracks[0] !== undefined) {
+        adaptation$.next(adaptation.trickModeTracks[0]);
+        return;
+      } else {
+        log.warn("TrackChoiceManager: No trickMode for video track" +
+                 "Playing video track instead.");
+        // disable trick mode when switching to a main video track
+        this._trickModeEnabled$.next(false);
```

Comment:
Oh ok that's why?
That's kind of ugly... I wouldn't expect the public api's private state to be mutated in another class.